include/swift/SIL/OptimizationRemark.h

```diff
@@ -75,23 +76,30 @@ template <typename DerivedT> class Remark {
   StringRef PassName;
 
   /// Textual identifier for the remark (single-word, camel-case). Can be used
-  /// by external tools reading the YAML output file for optimization remarks to
+  /// by external tools reading the output file for optimization remarks to
   /// identify the remark.
-  StringRef Identifier;
+  SmallString<32> Identifier;
 
   /// Source location for the diagnostics.
   SourceLoc Location;
 
   /// The function for the diagnostics.
   SILFunction *Function;
 
+  /// The demangled name of \p Function.
+  SmallString<64> DemangledFunctionName;
```

Comment:
I am fine if you fix this in a follow on patch. But can you standardize Remark's ivars to how the rest of SIL does this? We use camelCase for ivars. SILModule is from very early in Swift and got this wrong (I will fix it eventually). That being said, lets not spread the badness.

Reply:
I added a separate commit to the PR that does that for OptimizationRemark.